Named page 23682 (#24485)
mdn/content: files/en-us/web/css/@page/page/index.md

---

> title: page
> slug: Web/CSS/@page/page
> page-type: css-at-rule-property

Reviewer: There are `css-at-rule-descriptor` and `css-property` page types, but no `css-at-rule-property` page types. This is regular CSS property, associated with the @page, but not a descriptor. We need to move the page and have it listed as:

 ```suggestion
title: page
slug: Web/CSS/page
page-type: css-property
```

Author: This makes sense to me now thank you

tags:
  - "@page"
  - At-rule property
  - CSS
  - CSS Property
  - NeedsBrowserCompatibility
  - Reference
  - Web

> browser-compat: css.at-rules.page.page

Reviewer: ```suggestion
browser-compat: css.properties.page
```

I don't know if this exists.

Reviewer 2: I don't think we are speaking about the CSS property here, but from an at-rule descriptor (or at-rule property, I think the terminology varies from one spec to another).

I may be mistaken, and this is worth clarifying.

Author: Yes @teoli2003 is correct this is a property specific to the page at-rule
https://drafts.csswg.org/css-page-3/#using-named-pages
The at-rule allows a name, which can then be referenced via the page property.
[Referenced in this BCD file](https://github.com/CodeRedDigital/browser-compat-data/blob/named-page-23682/css/at-rules/page.json#L47-L81)

---

{{CSSRef}}

> The **`page`** [CSS](/en-US/docs/Web/CSS) [at-rule](/en-US/docs/Web/CSS/At-rule) property, used with the {{cssxref("@page")}} at-rule. It is used to use a particular type of page called a **named page** that has been declared on a `@page` at-rule.

Reviewer: ```suggestion
The **`page`** [CSS](/en-US/docs/Web/CSS) property is used to specify the named page, a specific type of page defined by the {{cssxref("@page")}}  [at-rule](/en-US/docs/Web/CSS/At-rule).
```

> If there are multiple selectors that are using a named page consecutively then a forced page break using [`break-after`](/en-US/docs/Web/CSS/break-after) is needed.

Reviewer: is it required, or should we write: 
```suggestion
If there are multiple selectors that are using a named page consecutively then a forced page break using [`break-after`](/en-US/docs/Web/CSS/break-after) may be needed.
```

## Syntax

```css

> /* default values */
> page: auto;
> /* named page value using <custom-ident> */
> page: exampleName;

Reviewer: ```suggestion
/* set a named page */
page: exampleName;
page: chapterIntro;

/* Use ancestors named page */
page: auto; /* default value */

/* Global values */
page: inherit;
page: initial;
page: revert;
page: revert-layer;
page: unset;
```

I assume the global values are supported...

```

### Values

- `auto`

>   - : Default setting for the property.

Reviewer: ```suggestion
  - : Default value.  Use the value of the nearest ancestor with a non-`auto` value. If no ancestor has a named page value set, the empty string.
```

Author: Done

> - `name`

Reviewer: ```suggestion
- {{cssxref("<custom-ident>")}}
```

>   - : A value defined by a user on that is applied on the [`@page`](/en-US/docs/Web/CSS/@page) at-rule, this is case-sensitive and follows the syntax of [`<custom-ident>`](/en-US/docs/Web/CSS/custom-ident).

Reviewer: ```suggestion
  - : Case-sensitive name defined in a [`@page`](/en-US/docs/Web/CSS/@page) at-rule.
```

> > **Note:** The page property does not inherit. However, if the page value on an element is auto, then its used value is the value specified on its nearest ancestor with a non-auto value. When specified on the root element, the used value for auto is the empty string.

Reviewer: ```suggestion
> **Note:** The `page` property does not inherit. However, if the `page` value on an element is `auto`, then its used value is the value specified on its nearest ancestor with a non-`auto` value. When specified on the root element, the used value for auto is the empty string.
```

Reviewer: I added this content as a suggested edit for `auto`, so maybe needs rework or omission. But if important, the note is ok too.

Author: I updated the `auto` description and remooved this note.
